src/extension/commands/index.ts

```diff
@@ -6,22 +6,44 @@ import {
   workspace, NotebookData, commands, NotebookCellData, NotebookCellKind
 } from 'vscode'
 
-import { getBinaryPath, getTLSDir, getTLSEnabled } from '../../utils/configuration'
+import { getBinaryPath, getTLSDir, getTLSEnabled, isNotebookTerminalEnabledForCell } from '../../utils/configuration'
 import { Kernel } from '../kernel'
-import { getAnnotations, getTerminalByCell } from '../utils'
+import { getAnnotations, getTerminalByCell, replaceOutput } from '../utils'
 import RunmeServer from '../server/runmeServer'
 import { GrpcRunnerEnvironment } from '../runner'
 
 function showWarningMessage () {
   return window.showWarningMessage('Couldn\'t find terminal! Was it already closed?')
 }
 
-export function openTerminal (cell: NotebookCell) {
-  const terminal = getTerminalByCell(cell)
-  if (!terminal) {
-    return showWarningMessage()
+export function openTerminal (kernel: Kernel, grpcRunner: boolean) {
+  return async function (cell: NotebookCell) {
+    const terminal = getTerminalByCell(cell)
+    if (!terminal) {
+      return showWarningMessage()
+    }
+
+    if (isNotebookTerminalEnabledForCell(cell) && grpcRunner) {
+      const terminalOutput = kernel.getCellTerminalOutputPayload(cell)
+
+      if (terminalOutput) {
+        let exec
+        try {
+          exec = await kernel.createCellExecution(cell)
+          exec.start(Date.now())
+
+          await replaceOutput(exec, terminalOutput)
+        } catch (e: any) {
+          window.showErrorMessage(e.message)
+        } finally {
+          exec?.end(true)
+          return
```

Comment:
This logic could be moved to a method exposed from here https://github.com/stateful/vscode-runme/blob/main/src/extension/notebookTerminal.ts

Reply:
Yeah, I removed that class in this PR.

I'm happy to add it back at some point, but I felt like it was silly having an abstraction which just wraps one static method.